### ISSUE TITLE
Better way to handle unmatched IP versions on service and endpoint

### ISSUE THREAD

**Issue by MrHohn** [kubernetes/kubernetes#60471]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network

**What happened**:
Ref https://github.com/kubernetes/kubernetes/issues/57219, lately we found an issue that a service/endpoint could possibly have unmatched IP versions configured (e.g. some fields are in IPv4 while others are IPv6, or uses ipv6 while cluster is in ipv4), which causes kube-proxy malfunctioning.

https://github.com/kubernetes/kubernetes/pull/56880 plumbs some detection codes to protect kube-proxy, however, the fact that we have to add logic into kube-proxy is not desired. kube-proxy wouldn't need to handle this if "mis-configured" service/endpoint is invalidated earlier on the apiserver side. Besides, the definition of "mis-configured" is yet to be defined --- probably in some cases have both ipv4 and ipv6 configs makes sense?

@freehan also had a quick thought on adding an admission controller for this.

In short, what we are adding now (https://github.com/kubernetes/kubernetes/pull/56880) is not great and would be great to have further discussion on this.
